ocaml/lambda/translcore.ml

```diff
@@ -32,45 +32,31 @@ type error =
   | Unreachable_reached
   | Bad_probe_layout of Ident.t
   | Non_value_layout of Layout.Violation.t
+  | Void_sort of type_expr
 
 exception Error of Location.t * error
 
 let use_dup_for_constant_mutable_arrays_bigger_than = 4
 
-(* CR layouts v2: When we're ready to allow non-values, these can be deleted or
-   changed to check for void. *)
-let sort_must_be_value ~why loc sort =
-  if not Sort.(equate sort value) then
-    let violation = Layout.(Violation.of_ (Not_a_sublayout
-                                             (of_sort ~why sort,
-                                              value ~why:V1_safety_check))) in
-    raise (Error (loc, Non_value_layout violation))
-
 let layout_must_be_value loc layout =
   match Layout.(sub layout (value ~why:V1_safety_check)) with
   | Ok _ -> ()
   | Error e -> raise (Error (loc, Non_value_layout e))
 
-(* CR layouts v2: In the places where this is used, we want to allow any (the
-   left of a semicolon and loop bodies).  So we want this instead of the usual
-   sanity check for value.  But we still default to value before checking for
-   void, to allow for sort variables arising in situations like
+(* CR layouts v2: In the places where this is used, we will want to allow
```

Comment:
Is there any reason we want to disallow void in such places? It seems to me there are no more complexities about allowing `#float` than about allowing void.

Reply:
The main issue is that support for void requires more thought about front-end and transl support for things like record projection.  Once we have types with different physical sizes, the way offsets for operations like this are communicated from the front-end to later phases of the compiler need more careful testing.

I know you folks are well ahead of us on things like unarization, but even if that's ready to go we are trying to be very cautious and ship very incrementally.

Reply:
Ah I see. For now in the middle and back end, we assume anything that is stored in a block is a value, but the rest can be non-values.